### DISCUSSION TITLE
How to backup and restore mysql database?

### DISCUSSION THREAD

**Discussion by sergeymeleschenko** [DefectDojo/django-DefectDojo#6726] [General]
I need to backup MySQL database from defectdojo version 2.9.0 and restore it to another machine with defectdojo version 2.13.1.
How to backup mysql database, if there some manual about this ? Where can I find info.

**Comment by mtesauro** [comment]
The easiest thing to do is take a sqldump of the DB on 2.9.0 and then restore using that dump.  A quick search will find loads of articles on how to do that for MySQL like [this one](https://www.sqlshack.com/how-to-backup-and-restore-mysql-databases-using-the-mysqldump-command/).

When you startup 2.13.1 make sure you run the migrations. If you're using docker-compose, this will 'just happen' with the initializer container.  You may want to use `docker logs -f` to watch the migrations on that container to ensure there aren't errors since you're doing a rather large version update.

**Reply by sergeymeleschenko** [reply]
I'm doing backup on the 1 machine with defefctdojo 2.9 and restoring to the 2nd machine with defectdojo version 2.13.1. run the migrations - what do you meen ? how can I run it ?

**Reply by sergeymeleschenko** [reply]
![image](https://user-images.githubusercontent.com/34446179/185153924-3c88da8e-f74b-43c0-a17c-437c3dba561b.png)

**Reply by mtesauro** [reply]
Need to know what type of install you have before any more details can be provided.

gododoj (iron) or docker-compose/k8s (container-based) have different ways to run the database migrations (and ensure they ran without errors)

**Comment by sergeymeleschenko** [comment]
Docker-compose on both machines. ./dc-up-d mysql-rabbitmq .

ср, 17 серп. 2022, 19:18 користувач Matt Tesauro ***@***.***>
пише:

**Comment by mtesauro** [comment]
So, for docker-compose, you'll need to do start the compose and do:

`docker logs -f initializer `
or 
`docker-compose logs -f initializer` ([docs](https://docs.docker.com/engine/reference/commandline/compose_logs/))

to watch the initializer container's output - it will run the migrations and show any errors that occur.

If the logs don't show any errors, then I'd check that the migrations all worked (more info [here](https://docs.djangoproject.com/en/4.1/ref/django-admin/#django-admin-showmigrations))

You can see what migrations ran with:

`docker exec -it uwsgi bash -c "python manage.py showmigrations"`

The output should looks something like:

```
$ docker exec -it uwsgi bash -c "python manage.py showmigrations"
[18/Aug/2022 03:01:13] INFO [dojo.models:4002] enabling audit logging
admin
 [X] 0001_initial
 [X] 0002_logentry_remove_auto_add
 [X] 0003_logentry_add_action_flag_choices
auditlog
 [X] 0001_initial
 [X] 0002_auto_support_long_primary_keys
 [X] 0003_logentry_remote_addr
         <<MANY LINES REMOVED>>
tagging
 [X] 0001_initial
 [X] 0002_on_delete
 [X] 0003_adapt_max_tag_length
watson
 [X] 0001_initial
```
Basically every line for a migration should have a '[X]' at the beginning. If an 'X" is missing, that migration failed/didn't run.

**Reply by sergeymeleschenko** [reply]
admin
 [X] 0001_initial
 [X] 0002_logentry_remove_auto_add
 [X] 0003_logentry_add_action_flag_choices
auditlog
 [X] 0001_initial
 [X] 0002_auto_support_long_primary_keys
 [X] 0003_logentry_remote_addr
 [X] 0004_logentry_detailed_object_repr
 [X] 0005_logentry_additional_data_verbose_name
 [X] 0006_object_pk_index
 [X] 0007_object_pk_type
 [X] 0008_action_index
 [X] 0009_alter_logentry_additional_data
 [X] 0010_alter_logentry_timestamp
auth
 [X] 0001_initial
 [X] 0002_alter_permission_name_max_length
 [X] 0003_alter_user_email_max_length
 [X] 0004_alter_user_username_opts
 [X] 0005_alter_user_last_login_null
 [X] 0006_require_contenttypes_0002
 [X] 0007_alter_validators_add_error_messages
 [X] 0008_alter_user_username_max_length
 [X] 0009_alter_user_last_name_max_length
 [X] 0010_alter_group_name_max_length
 [X] 0011_update_proxy_permissions
 [X] 0012_alter_user_first_name_max_length
authtoken
 [X] 0001_initial
 [X] 0002_auto_20160226_1747
 [X] 0003_tokenproxy
contenttypes
 [X] 0001_initial
 [X] 0002_remove_content_type_name
django_celery_results
 [X] 0001_initial
 [X] 0002_add_task_name_args_kwargs
 [X] 0003_auto_20181106_1101
 [X] 0004_auto_20190516_0412
 [X] 0005_taskresult_worker
 [X] 0006_taskresult_date_created
 [X] 0007_remove_taskresult_hidden
 [X] 0008_chordcounter
 [X] 0009_groupresult
 [X] 0010_remove_duplicate_indices
 [X] 0011_taskresult_periodic_task_name
dojo
 [X] 0001_squashed_0090_index_duplicate_finding (90 squashed migrations)
 [X] 0091_npm_audit_path_censoring
 [X] 0092_is_mitigated
 [X] 0093_django_tagging_removal
 [X] 0094_remove_system_settings_s_finding_severity_naming
 [X] 0095_remove_old_product_contact_fields
 [X] 0096_grype_name_change
 [X] 0097_engagement_type
 [X] 0098_anchore_vuln_id
 [X] 0099_delete_report
 [X] 0100_dojo_user_for_authv2
 [X] 0101_enable_features
 [X] 0102_dojo_group
 [X] 0103_report_notification
 [X] 0104_endpoint_userinfo_creation
 [X] 0105_endpoint_host_migration
 [X] 0106_role_model
 [X] 0107_global_role
 [X] 0108_blank_fields
 [X] 0109_group_user_role
 [X] 0110_auth_v2_migrate_user_roles
 [X] 0111_group_user_rename
 [X] 0112_group_user_rename_2
 [X] 0113_endpoint_protocol
 [X] 0114_cyclonedx_vuln_uniqu
 [X] 0115_language_types
 [X] 0116_test_type_active
 [X] 0117_usercontactinfo_force_password_reset
 [X] 0118_remove_finding_images
 [X] 0119_default_group_is_staff
 [X] 0120_sonarqube_test_and_clean
 [X] 0121_user_restrict
 [X] 0122_cobaltio_product
 [X] 0123_scan_type
 [X] 0124_sonarqube_api_type_length_change
 [X] 0125_sonarqube_clean
 [X] 0126_finding_publish_date
 [X] 0127_remove_hashes
 [X] 0128_pytz_update
 [X] 0129_finding_deprecated_fields
 [X] 0130_product_api_scan_configuration
 [X] 0131_migrate_sonarcube_cobalt
 [X] 0132_remove_configs_from_test
 [X] 0133_finding_service
 [X] 0134_sonarque_cobaltio_removal
 [X] 0135_email_from
 [X] 0136_default_group_help_text
 [X] 0137_system_settings_enable_endpoint_metadata_import
 [X] 0138_remove_authorized_users
 [X] 0139_google_sheets_rules_framework_enable
 [X] 0140_auth_group
 [X] 0141_enable_user_profile_editable
 [X] 0142_environment_delete
 [X] 0143_objects
 [X] 0144_import_action_untouched
 [X] 0145_system_settings_default_group_email_pattern
 [X] 0146_lead_optional
 [X] 0147_rename_sslyze_parser
 [X] 0148_default_notifications
 [X] 0149_harmonize_user_format
 [X] 0150_dedupe_endpoint_status
 [X] 0151_index_endpoint_status
 [X] 0152_notifications_template
 [X] 0153_migrate_endpoint_mitigated
 [X] 0154_remove_endpoint_mitigated
 [X] 0155_enable_finding_groups
 [X] 0156_migrate_finding_groups_setting
 [ ] 0157_vulnerability_reference
 [ ] 0158_vulnerability_id
 [ ] 0159_remove_broken_endpoint_statuses
 [ ] 0160_set_notnull_endpoint_statuses
 [ ] 0161_alter_dojo_group_social_provider
 [ ] 0162_created_and_updated
 [ ] 0163_system_settings_enable_calendar
 [ ] 0164_remove_system_settings_staff_user_email_pattern
sessions
 [X] 0001_initial
sites
 [X] 0001_initial
 [X] 0002_alter_domain_unique
social_django
 [X] 0001_initial (2 squashed migrations)
 [X] 0002_add_related_name (2 squashed migrations)
 [X] 0003_alter_email_max_length (2 squashed migrations)
 [X] 0004_auto_20160423_0400 (2 squashed migrations)
 [X] 0005_auto_20160727_2333 (1 squashed migrations)
 [X] 0006_partial
 [X] 0007_code_timestamp
 [X] 0008_partial_timestamp
 [X] 0009_auto_20191118_0520
 [X] 0010_uid_db_index
tagging
 [X] 0001_initial
 [X] 0002_on_delete
 [X] 0003_adapt_max_tag_length
watson
 [X] 0001_initial

**Reply by sergeymeleschenko** [reply]
docker-compose logs -f initializer 
docker-compose logs -f initializer
WARN[0000] The "DD_DATABASE_HOST" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_PORT" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_URL" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_CELERY_BROKER_URL" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_NAME" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_USER" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_PASSWORD" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_URL" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_CELERY_BROKER_URL" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_HOST" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_PORT" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_HOST" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_PORT" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_URL" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_CELERY_BROKER_URL" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_HOST" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_PORT" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_URL" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_PASSWORD" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_NAME" variable is not set. Defaulting to a blank string.
WARN[0000] The "DD_DATABASE_USER" variable is not set. Defaulting to a blank string.
django-defectdojo-initializer-1  | wait-for-it.sh: waiting max 60 seconds for mysql:3306
django-defectdojo-initializer-1  | wait-for-it.sh: mysql:3306 is available after 0 seconds
django-defectdojo-initializer-1  | ============================================================
django-defectdojo-initializer-1  |      Overriding DefectDojo's local_settings.py with multiple
django-defectdojo-initializer-1  |      Files: /app/docker/extra_settings/README.md
django-defectdojo-initializer-1  | ============================================================
django-defectdojo-initializer-1  | Initializing.
django-defectdojo-initializer-1  | [17/Aug/2022 13:56:35] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | Waiting for database to be reachable
django-defectdojo-initializer-1  | Making migrations
django-defectdojo-initializer-1  | [17/Aug/2022 13:56:39] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | No changes detected in app 'dojo'
django-defectdojo-initializer-1  | Migrating
django-defectdojo-initializer-1  | [17/Aug/2022 13:56:54] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | Operations to perform:
django-defectdojo-initializer-1  |   Apply all migrations: admin, auditlog, auth, authtoken, contenttypes, django_celery_results, dojo, sessions, sites, social_django, tagging, watson
django-defectdojo-initializer-1  | Running migrations:
django-defectdojo-initializer-1  |   Applying auditlog.0010_alter_logentry_timestamp... OK
django-defectdojo-initializer-1  | Traceback (most recent call last):
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
django-defectdojo-initializer-1  |     return self.cursor.execute(sql, params)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 73, in execute
django-defectdojo-initializer-1  |     return self.cursor.execute(query, args)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 206, in execute
django-defectdojo-initializer-1  |     res = self._query(query)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 319, in _query
django-defectdojo-initializer-1  |     db.query(q)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/connections.py", line 254, in query
django-defectdojo-initializer-1  |     _mysql.connection.query(self, query)
django-defectdojo-initializer-1  | MySQLdb.OperationalError: (1022, "Can't write; duplicate key in table '#sql-1_13'")
django-defectdojo-initializer-1  |
django-defectdojo-initializer-1  | The above exception was the direct cause of the following exception:
django-defectdojo-initializer-1  |
django-defectdojo-initializer-1  | Traceback (most recent call last):
django-defectdojo-initializer-1  |   File "manage.py", line 11, in <module>
django-defectdojo-initializer-1  |     execute_from_command_line(sys.argv)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
django-defectdojo-initializer-1  |     utility.execute()
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
django-defectdojo-initializer-1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
django-defectdojo-initializer-1  |     self.execute(*args, **cmd_options)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
django-defectdojo-initializer-1  |     output = self.handle(*args, **options)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
django-defectdojo-initializer-1  |     res = handle_func(*args, **kwargs)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 244, in handle
django-defectdojo-initializer-1  |     post_migrate_state = executor.migrate(
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
django-defectdojo-initializer-1  |     state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
django-defectdojo-initializer-1  |     state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 230, in apply_migration
django-defectdojo-initializer-1  |     migration_recorded = True
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 118, in __exit__
django-defectdojo-initializer-1  |     self.execute(sql)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 145, in execute
django-defectdojo-initializer-1  |     cursor.execute(sql, params)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
django-defectdojo-initializer-1  |     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
django-defectdojo-initializer-1  |     return executor(sql, params, many, context)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
django-defectdojo-initializer-1  |     return self.cursor.execute(sql, params)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
django-defectdojo-initializer-1  |     raise dj_exc_value.with_traceback(traceback) from exc_value
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
django-defectdojo-initializer-1  |     return self.cursor.execute(sql, params)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 73, in execute
django-defectdojo-initializer-1  |     return self.cursor.execute(query, args)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 206, in execute
django-defectdojo-initializer-1  |     res = self._query(query)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 319, in _query
django-defectdojo-initializer-1  |     db.query(q)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/connections.py", line 254, in query
django-defectdojo-initializer-1  |     _mysql.connection.query(self, query)
django-defectdojo-initializer-1  | django.db.utils.OperationalError: (1022, "Can't write; duplicate key in table '#sql-1_13'")
django-defectdojo-initializer-1  |   Applying dojo.0157_vulnerability_reference...Admin user: admin
django-defectdojo-initializer-1  | [17/Aug/2022 13:57:08] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | Admin password: Initialization detected that the admin user admin already exists in your database.
django-defectdojo-initializer-1  | If you don't remember the admin password, you can create a new superuser with:
django-defectdojo-initializer-1  | $ docker-compose exec uwsgi /bin/bash -c 'python manage.py createsuperuser'
django-defectdojo-initializer-1  | Initialization of test_types
django-defectdojo-initializer-1  | [17/Aug/2022 13:57:09] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | Creation of non-standard permissions
django-defectdojo-initializer-1  | [17/Aug/2022 13:57:11] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | [17/Aug/2022 13:57:12] INFO [dojo.management.commands.initialize_permissions:28] Non-standard permissions have been created
django-defectdojo-initializer-1  | wait-for-it.sh: waiting max 60 seconds for mysql:3306
django-defectdojo-initializer-1  | wait-for-it.sh: mysql:3306 is available after 2 seconds
django-defectdojo-initializer-1  | ============================================================
django-defectdojo-initializer-1  |      Overriding DefectDojo's local_settings.py with multiple
django-defectdojo-initializer-1  |      Files: /app/docker/extra_settings/README.md
django-defectdojo-initializer-1  | ============================================================
django-defectdojo-initializer-1  | Initializing.
django-defectdojo-initializer-1  | [19/Aug/2022 07:57:42] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | Waiting for database to be reachable
django-defectdojo-initializer-1  | Making migrations
django-defectdojo-initializer-1  | [19/Aug/2022 07:57:47] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | No changes detected in app 'dojo'
django-defectdojo-initializer-1  | Migrating
django-defectdojo-initializer-1  | [19/Aug/2022 07:58:09] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | Operations to perform:
django-defectdojo-initializer-1  |   Apply all migrations: admin, auditlog, auth, authtoken, contenttypes, django_celery_results, dojo, sessions, sites, social_django, tagging, watson
django-defectdojo-initializer-1  | Running migrations:
django-defectdojo-initializer-1  | Traceback (most recent call last):
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 82, in _execute
django-defectdojo-initializer-1  |     return self.cursor.execute(sql)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 73, in execute
django-defectdojo-initializer-1  |     return self.cursor.execute(query, args)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 206, in execute
django-defectdojo-initializer-1  |     res = self._query(query)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 319, in _query
django-defectdojo-initializer-1  |     db.query(q)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/connections.py", line 254, in query
django-defectdojo-initializer-1  |     _mysql.connection.query(self, query)
django-defectdojo-initializer-1  | MySQLdb.OperationalError: (1050, "Table 'dojo_vulnerability_reference' already exists")
django-defectdojo-initializer-1  |
django-defectdojo-initializer-1  | The above exception was the direct cause of the following exception:
django-defectdojo-initializer-1  |
django-defectdojo-initializer-1  | Traceback (most recent call last):
django-defectdojo-initializer-1  |   File "manage.py", line 11, in <module>
django-defectdojo-initializer-1  |     execute_from_command_line(sys.argv)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
django-defectdojo-initializer-1  |     utility.execute()
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
django-defectdojo-initializer-1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
django-defectdojo-initializer-1  |     self.execute(*args, **cmd_options)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
django-defectdojo-initializer-1  |     output = self.handle(*args, **options)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
django-defectdojo-initializer-1  |     res = handle_func(*args, **kwargs)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 244, in handle
django-defectdojo-initializer-1  |     post_migrate_state = executor.migrate(
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
django-defectdojo-initializer-1  |     state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
django-defectdojo-initializer-1  |     state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
django-defectdojo-initializer-1  |     state = migration.apply(state, schema_editor)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/migration.py", line 126, in apply
django-defectdojo-initializer-1  |     operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/migrations/operations/models.py", line 92, in database_forwards
django-defectdojo-initializer-1  |     schema_editor.create_model(model)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 345, in create_model
django-defectdojo-initializer-1  |     self.execute(sql, params or None)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 145, in execute
django-defectdojo-initializer-1  |     cursor.execute(sql, params)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
django-defectdojo-initializer-1  |     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
django-defectdojo-initializer-1  |     return executor(sql, params, many, context)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
django-defectdojo-initializer-1  |     return self.cursor.execute(sql, params)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
django-defectdojo-initializer-1  |     raise dj_exc_value.with_traceback(traceback) from exc_value
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 82, in _execute
django-defectdojo-initializer-1  |     return self.cursor.execute(sql)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/django/db/backends/mysql/base.py", line 73, in execute
django-defectdojo-initializer-1  |     return self.cursor.execute(query, args)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 206, in execute
django-defectdojo-initializer-1  |     res = self._query(query)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 319, in _query
django-defectdojo-initializer-1  |     db.query(q)
django-defectdojo-initializer-1  |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/connections.py", line 254, in query
django-defectdojo-initializer-1  |     _mysql.connection.query(self, query)
django-defectdojo-initializer-1  | django.db.utils.OperationalError: (1050, "Table 'dojo_vulnerability_reference' already exists")
django-defectdojo-initializer-1  |   Applying dojo.0157_vulnerability_reference...Admin user: admin
django-defectdojo-initializer-1  | [19/Aug/2022 07:58:15] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | Admin password: Initialization detected that the admin user admin already exists in your database.
django-defectdojo-initializer-1  | If you don't remember the admin password, you can create a new superuser with:
django-defectdojo-initializer-1  | $ docker-compose exec uwsgi /bin/bash -c 'python manage.py createsuperuser'
django-defectdojo-initializer-1  | Initialization of test_types
django-defectdojo-initializer-1  | [19/Aug/2022 07:58:16] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | Creation of non-standard permissions
django-defectdojo-initializer-1  | [19/Aug/2022 07:58:19] INFO [dojo.models:4002] enabling audit logging
django-defectdojo-initializer-1  | [19/Aug/2022 07:58:20] INFO [dojo.management.commands.initialize_permissions:28] Non-standard permissions have been created
